vnext/Shared/Modules/HttpModule.cpp

```diff
@@ -0,0 +1,192 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+
+#include "HttpModule.h"
+
+// React Native
+#include <cxxreact/Instance.h>
+#include <cxxreact/JsArgumentHelpers.h>
+
+using facebook::react::Instance;
+using folly::dynamic;
+using std::shared_ptr;
+using std::string;
+using std::weak_ptr;
+
+namespace {
+
+using Microsoft::React::IHttpResource;
+
+constexpr char moduleName[] = "Networking";
+
+// TODO: Add to shared header? (See WebSocketModule)
+static void SendEvent(weak_ptr<Instance> weakInstance, string &&eventName, dynamic &&args) {
+  if (auto instance = weakInstance.lock()) {
+    instance->callJSFunction("RCTDeviceEventEmitter", "emit", dynamic::array(std::move(eventName), std::move(args)));
+  }
+}
+
+static shared_ptr<IHttpResource> CreateHttpResource(weak_ptr<Instance> weakInstance) {
+  auto resource = IHttpResource::Make();
+
+  resource->SetOnResponse([weakInstance](int64_t requestId, IHttpResource::Response &&response) {
+    dynamic headers = dynamic::object();
+    for (auto &header : response.Headers) {
+      headers[header.first] = header.second;
+    }
+
+    // TODO: Test response content.
+    dynamic args = dynamic::array(requestId, response.StatusCode, headers, response.Url);
+
+    SendEvent(weakInstance, "didReceiveNetworkResponse", std::move(args));
+  });
+
+  resource->SetOnData([weakInstance](int64_t requestId, std::string &&responseData) {
+    dynamic args = dynamic::array(requestId, std::move(responseData));
+
+    SendEvent(weakInstance, "didReceiveNetworkData", std::move(args));
+
+    // TODO: Move into separate method IF not executed right after onData()
+    SendEvent(weakInstance, "didCompleteNetworkResponse", dynamic::array(requestId));
+  });
+
+  resource->SetOnError([weakInstance](int64_t requestId, string &&message) {
+    dynamic args = dynamic::array(requestId, std::move(message));
+    // TODO: isTimeout errorArgs.push_back(true);
+
+    SendEvent(weakInstance, "didCompleteNetworkResponse", std::move(args));
+  });
+
+  return resource;
+}
+
+} // namespace
+
+namespace Microsoft::React {
+
+HttpModule::HttpModule() noexcept : m_holder{std::make_shared<ModuleHolder>()} {
+  m_holder->Module = this;
```

Comment:
>this [](http://example.com/codeflow?start=21&length=4)

Isn't this a circular reference?

Reply:
Sort of. This is needed, though, to be able to retrieve the React Instance weak pointer at the proper stage (when `getMethods()` is called).

This is due to a bug in the core React Native code (module's Instance being set AFTER calling `getMethods()`) which has been fixed recently, but such fix won't be available in earlier React Native releases (i.e. 0.66, 0.67).

I implemented this pattern based on @vmoroz's suggestion.

Reply:
See `WebSocketModule` for reference. The same pattern was applied and it prevents race condition-induced access violation on tear-down scenarios when there are pending callbacks.